src/palace/manager/util/settings.py

```diff
@@ -0,0 +1,134 @@
+import functools
+from collections.abc import Callable
+from typing import Any
+
+from pydantic.env_settings import BaseSettings, SettingsSourceCallable
+from pydantic.fields import ModelField
+
+from palace.manager.core.config import CannotLoadConfiguration
+from palace.manager.service.configuration import ServiceConfiguration
+from palace.manager.util.log import LoggerMixin
+
+
+class ServiceConfigurationWithLimitedEnvOverride(ServiceConfiguration, LoggerMixin):
```

Comment:
Would it make sense for this to live in `palace.manager.service.configuration` with the parent `ServiceConfiguration` class. Rather then in `palace.manager.util.settings`?

Reply:
Yeah. Probably. I wasn't entirely sure where to put it, but that makes sense.

Reply:
Though, as I noted above, it cannot live in the same file:
> Not sure that this is the best place for this, but it cannot go in the same file with ServiceConfiguration because of an import loop due to LoggerMixin.

Reply:
Maybe we could move `palace.manager.service.configuration.ServiceConfiguration` to `palace.manager.service.configuration.base.ServiceConfiguration` and put this in the `palace.manager.service.configuration` package. We can just leave it where it is for now as well.

Reply:
@jonathangreen I didn't see this before I moved them. I put both of them in `palace.manager.service.configuration` package directory. I can move them, though, if you'd prefer.

Reply:
No whats here looks good to me.